contributing.md

```diff
@@ -0,0 +1,135 @@
+# Contribution
+
+Thanks for contributing to Trunk's default plugins! Read on to learn more.
+
+- [Overview](#overview)
+- [Release process](#releases)
+- [Adding new linters](#linters)
+- [Adding new actions](#actions)
+- [Guidelines](#guidelines)
+- [Docs](https://docs.trunk.io)
+
+## Overview
+
+We use this repository to provide our users with default linters and actions. trunk automatically
+adds the following to users' trunk.yaml:
+
+```yaml
+plugins:
+  sources:
+    - id: trunk
+      uri: https://github.com/trunk-io/plugins
+      ref: v0.0.8
+```
+
+Plugins can also be loaded from local paths, as below:
+
+```yaml
+plugins:
+  sources:
+    - id: trunk
+      local: /path/to/repo/root
+```
+
+Run `trunk run toggle-local` to quickly toggle this setting.
+
+Adding a plugin source lets users run `trunk check enable` or `trunk actions enable` with linters
+and actions defined in that plugin. For more information, see our
+[docs](https://docs.trunk.io/docs/plugins).
+
+If you want to request a new feature or get help with a problem, please see our
+[Features Page](https://features.trunk.io/) or our [Slack](https://slack.trunk.io/).
+
+## Releases
+
+Plugin sources are tied to the version specified with the `ref` field. When users run
+`trunk upgrade`, 3 fields in their trunk.yaml will be updated:
+
+1. The CLI version will be updated to the latest trunk binary.
+2. The plugin `ref` will be updated to the latest compatible release of this repo.
+3. All enabled linters will be updated to their latest versions.
+
+We aim to provide a fairly regular release cadence for this repository. Releases are tied to minimum
+compatible versions of the trunk CLI, identified by the `required_trunk_version` field. Users will
+not be able to load a plugin source until they have upgraded to a sufficient CLI version.
+
+If you need a linter or action definition from this repo that hasn't been released yet, you can
+change the `ref` field of a plugin source to be any `git` ref. However, we recommend sticking to our
+official tagged releases unless absolutely necessary.
+
+## Linters
+
+If you want to add a new linter integration with trunk, refer to our docs on
+[custom linters](https://docs.trunk.io/docs/check-custom-linters) and
+[custom parsers](https://docs.trunk.io/docs/custom-parsers), and work through the following:
```

Comment:
```suggestion
To add a new linter:
```

"If you want to" and "with trunk" are both redundant here: if a user is in these docs, they want to do so; and these docs are all in the context of trunk.